From: Duncan Eastoe <duncan.eastoe@att.com>
Date: Tue, 14 Jul 2020 18:19:35 +0100
Subject: tools: frr-reload: catch and log VtyshMarkException

Catch a VtyshMarkException returned from Config.load_from_file() when
loading the new configuration, log it, and exit non-zero. This means
we get more useful output ie. that from vtysh, rather than a stack
trace.

Signed-off-by: Duncan Eastoe <duncan.eastoe@att.com>
---
 tools/frr-reload.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/tools/frr-reload.py b/tools/frr-reload.py
index 1b2e9b3..53c2866 100755
--- a/tools/frr-reload.py
+++ b/tools/frr-reload.py
@@ -1267,7 +1267,11 @@ if __name__ == '__main__':
 
     # Create a Config object from the config generated by newconf
     newconf = Config()
-    newconf.load_from_file(args.filename, args.bindir, args.confdir)
+    try:
+        newconf.load_from_file(args.filename, args.bindir, args.confdir)
+    except VtyshMarkException as ve:
+        log.error(ve.output.decode())
+        sys.exit(1)
     reload_ok = True
 
     if args.test:
